app/labor/cache_buster.rb

```diff
@@ -19,17 +19,50 @@ def self.bust(path)
     #
     # https://github.com/fastly/fastly-ruby#efficient-purging
     return unless Rails.env.production?
-    return if ApplicationConfig["FASTLY_API_KEY"].blank?
 
-    HTTParty.post("https://api.fastly.com/purge/https://#{ApplicationConfig['APP_DOMAIN']}#{path}",
-                  headers: { "Fastly-Key" => ApplicationConfig["FASTLY_API_KEY"] })
-    HTTParty.post("https://api.fastly.com/purge/https://#{ApplicationConfig['APP_DOMAIN']}#{path}?i=i",
-                  headers: { "Fastly-Key" => ApplicationConfig["FASTLY_API_KEY"] })
+    if fastly_enabled?
+      bust_fastly_cache(path)
+    elsif nginx_enabled?
+      bust_nginx_cache(path)
+    end
   rescue URI::InvalidURIError => e
     Rails.logger.error("Trying to bust cache of an invalid uri: #{e}")
     DatadogStatsClient.increment("cache_buster.invalid_uri", tags: ["path:#{path}"])
   end
 
+  def self.fastly_enabled?
+    ApplicationConfig["FASTLY_API_KEY"].present? && ApplicationConfig["FASTLY_SERVICE_ID"].present?
+  end
+
+  def self.nginx_enabled?
+    ApplicationConfig["OPENRESTY_PROTOCOL"].present? && ApplicationConfig["OPENRESTY_DOMAIN"].present?
+  end
```

Comment:
these two probably deserved to be surfaced somewhere global so that you can both here and in Purgeable check `Cache.fastly_enabled?`

Definitely not blocking